system/src/system_network_diagnostics.cpp

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (c) 2019 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platforms.h"
+
+// Mesh signal strength/quality have not been implemented
+#if (PLATFORM_ID != PLATFORM_XENON) && (PLATFORM_ID != PLATFORM_XENON_SOM)
```

Comment:
FYI we've renamed them in https://github.com/particle-iot/device-os/pull/1774, so after 1.2.0 branch is rebased on develop and this branch is rebase on 1.2.0, this will need to get fixed.

Reply:
updated